### ISSUE TITLE
Cache lock contention with concurrent clj-kondo processes (editor + pre-commit hook) causes unhandled exception

### ISSUE THREAD

**Issue by jramosg** [clj-kondo/clj-kondo#2795]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

v2026.01.19

**macro usage**

N/A

**platform**

JVM (also reproduced with native binary)

**editor**

Calva (VS Code): clj-kondo runs in the background via the editor integration

**problem**

When committing with a pre-commit hook that runs `clj-kondo --lint`, the lint fails with an "Unexpected error" if the editor's clj-kondo integration is linting the same files concurrently. The two processes compete for the `.clj-kondo` cache file lock; after 6 retries (~1.5s total backoff), clj-kondo throws a hard exception that propagates as an unhandled error rather than a recoverable warning.

```
Unexpected error. Please report an issue.
java.lang.Exception: Clj-kondo cache is locked by other thread or process.
    at clj_kondo.impl.cache$sync_cache.invokeStatic(cache.clj:202)
    at clj_kondo.core$run_BANG_.invokeStatic(core.clj:242)
    at clj_kondo.main$main.invokeStatic(main.clj:150)
    ...
```

**repro**

1. Open a Clojure project in an editor that runs clj-kondo on file changes (e.g. Calva, clojure-lsp)
2. Add a pre-commit hook that runs `clj-kondo --lint` on staged files
3. Save a file (triggering background lint) and immediately `git commit`
4. The pre-commit hook fails with the cache lock error above

The same scenario occurs with two pre-commit hooks both invoking clj-kondo (e.g. a local `clj-kondo --lint` hook alongside a Docker-based hook that mounts the project directory).

**config**

N/A. bug occurs with default configuration.

**expected behavior**

When the cache is locked by another clj-kondo process, linting should continue without the cache (degraded mode) and emit a warning to stderr rather than failing the entire lint run with an unhandled exception. Cache contention is a normal operational condition when an editor integration and a pre-commit hook run concurrently; it should not prevent commits.
